### ISSUE TITLE
"Error importing mediapipe: unhashable type: 'list' " when simply running "import mediapipe as mp"

### ISSUE THREAD

**Issue by hashimh4** [google-ai-edge/mediapipe#5815]
### OS Platform and Distribution

Windows 11 Home

### Compiler version

g++ (Rev10, Built by MSYS2 project) 12.2.0

### Programming Language and version

Python 3.9.0

### Installed using virtualenv? pip? Conda?(if python)

installed with pip in a virtual python environment

### MediaPipe version

0.10.20

### Bazel version

8.0.0

### Android AAR (if android)

None

### OpenCV version (if running on desktop)

4.10.0.84

### Describe the problem

"Error importing mediapipe: unhashable type: 'list' " when simply running "import mediapipe as mp"

### Complete Logs

```shell
Error importing mediapipe: unhashable type: 'list
```

**Comment by hashimh4** [comment]
Package                      Version
---------------------------- -----------
absl-py                      2.1.0
aiohappyeyeballs             2.4.4
aiohttp                      3.11.11
aiosignal                    1.3.2
astunparse                   1.6.3
async-timeout                5.0.1
attrs                        24.3.0
certifi                      2024.12.14
cffi                         1.17.1
charset-normalizer           3.4.1
cmake                        3.31.2
colorama                     0.4.6
contourpy                    1.3.0
cycler                       0.12.1
decorator                    4.4.2
efficientnet-pytorch         0.7.1
filelock                     3.16.1
flatbuffers                  24.12.23
fonttools                    4.55.3
frozenlist                   1.5.0
fsspec                       2024.12.0
future                       1.0.0
gast                         0.6.0
google-pasta                 0.2.0
grpcio                       1.69.0
h5py                         3.12.1
huggingface-hub              0.27.1
idna                         3.10
imageio                      2.36.1
imageio-ffmpeg               0.5.1
importlib_metadata           8.5.0
importlib_resources          6.5.2
jax                          0.4.30
jaxlib                       0.4.30
Jinja2                       3.1.5
keras                        3.8.0
kiwisolver                   1.4.7
libclang                     18.1.1
lightning-utilities          0.11.9
Markdown                     3.7
markdown-it-py               3.0.0
MarkupSafe                   3.0.2
matplotlib                   3.9.4
mdurl                        0.1.2
mediapipe                    0.10.20
mido                         1.3.3
mir-eval                     0.7
ml-dtypes                    0.4.0
moviepy                      1.0.3
mpmath                       1.3.0
multidict                    6.1.0
munch                        4.0.0
namex                        0.0.8
networkx                     3.2.1
numpy                        1.26.4
opencv-python                4.10.0.84
opt_einsum                   3.4.0
optree                       0.13.1
packaging                    24.2
pillow                       10.4.0
pip                          24.3.1
pretrainedmodels             0.7.4
pretty-midi                  0.2.10
proglog                      0.1.10
propcache                    0.2.1
protobuf                     4.25.5
pycparser                    2.22
pydub                        0.25.1
Pygments                     2.19.1
pyparsing                    3.2.1
python-dateutil              2.9.0.post0
python-dotenv                1.0.1
pytorch-lightning            2.5.0.post0
PyYAML                       6.0.2
requests                     2.32.3
rich                         13.9.4
safetensors                  0.5.2
scipy                        1.13.1
segmentation_models_pytorch  0.4.0
sentencepiece                0.2.0
setuptools                   75.8.0
six                          1.17.0
sounddevice                  0.5.1
sympy                        1.13.1
tensorboard                  2.18.0
tensorboard-data-server      0.7.2
tensorflow                   2.18.0
tensorflow_intel             2.18.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.5.0
timm                         1.0.12
torch                        2.5.1
torchaudio                   2.5.1
torchmetrics                 1.6.1
torchvision                  0.20.1
tqdm                         4.67.1
typing_extensions            4.12.2
urllib3                      2.3.0
Werkzeug                     3.1.3
wheel                        0.45.1
wrapt                        1.17.0
yarl                         1.18.3
zipp                         3.21.0

**Comment by kuaashish** [comment]
Hi @hashimh4,

Could you please outline the exact steps you are following, referencing the documentation? This information will help us reproduce and better understand the issue on our end.

Thank you!!

**Comment by hashimh4** [comment]
Hi @kuaashish,

In a new python file, the only code I run is:

try:
    import mediapipe as mp
    print("mediapipe imported successfully")
except Exception as e:
    print(f"Error importing mediapipe: {e}")

And it produces the error:
Error importing mediapipe: unhashable type: 'list

When adding the same block of code, only to try importing mediapipe and nothing else mediapipe related, with a large python file, then the same error is produced and the code terminated early.
Taking out the chunk of code makes the code run smoothly. Similarly simply import mediapipe as mp does not work and makes my code terminated early.

**Comment by hashimh4** [comment]
Hi @kuaashish,

In a new python file, the only code I run is:

try:
    import mediapipe as mp
    print("mediapipe imported successfully")
except Exception as e:
    print(f"Error importing mediapipe: {e}")

And it produces the error:
Error importing mediapipe: unhashable type: 'list

When adding the same block of code, only to try importing mediapipe and nothing else mediapipe related, with a large python file, then the same error is produced and the code terminated early.
Taking out the chunk of code makes the code run smoothly. Similarly simply import mediapipe as mp does not work and makes my code terminated early.

Edit: The issue persists after setting up a new virtual python environment. And adding the line also stops my code from completely running. Maybe there is a module conflict / version conflict? Please see the list previously

**Comment by kuaashish** [comment]
Hi @hashimh4,

Apologies for the delayed response. Are you still seeking a resolution for this? If so, the issue seems to be related to the virtual environment. Please allow us some time to reproduce the problem, and we will update you on the status shortly.

Thank you!!

**Comment by luisstrongest** [comment]
I have solved the problem by installing python 3.11, for which there is support for tensorflow @hashimh4
